### ISSUE TITLE
Flaky tests for Pekko Connectors Slick connector started occurring only with Slick 3.5.0-M4 on Scala 3

### ISSUE THREAD

**Issue by mdedetrich** [slick/slick#2811]
We are currently trying to add Scala 3 support to a Slick connector at Pekko and we are getting flaky test errors only for Scala 3, see https://github.com/apache/incubator-pekko-connectors/pull/223#issuecomment-1692537114. An example of a failing test is https://github.com/apache/incubator-pekko-connectors/actions/runs/5954495057/job/16152684518

Need to diagnose further what the problem is

**Comment by OlegYch** [comment]
i tried the pekko branch and couldn't reproduce the failure

**Comment by mdedetrich** [comment]
> i tried the [pekko ](https://github.com/pjfanning/incubator-pekko-connectors/tree/scala3-slick)branch and couldn't reproduce the failure 

I can't reproduce it locally, it's only occurring in CI. Typicly this wouldn't be a concern, however the critical thing is that it's only flaky for Scala 3 in CI and not for Scala 2 and that's the reason for creating this bug report.

**Comment by OlegYch** [comment]
i'd suggest trying with slick 3.4 binaries on scala3 to narrow down the cause

**Comment by mdedetrich** [comment]
> i'd suggest trying with slick 3.4 binaries on scala3 to narrow down the cause

From what I can tell the first (and for now only) Slick binaries that were released with Scala 3 is 3.5.0-M4 which is what that PR is using.

Are you saying that we should manually compile/publishLocal Slick 3.4 for Scala 3?

**Comment by OlegYch** [comment]
I mean trying with crossversion for3use213 https://www.scala-lang.org/blog/2021/04/08/scala-3-in-sbt.html#using-scala-213-libraries-in-scala-3

**Comment by mdedetrich** [comment]
> I mean trying with crossversion for3use213 https://www.scala-lang.org/blog/2021/04/08/scala-3-in-sbt.html#using-scala-213-libraries-in-scala-3

Got it, will do some digging to try and minimize this.

**Comment by nafg** [comment]
What is `docs.javadsl.User@b3f21ec9`? Maybe it would be more clear what it's not happy about and what is going wrong with a proper toString or otherwise adding more logging or assertions to the test to narrow down what about the expectation is broken

**Comment by mdedetrich** [comment]
> What is `docs.javadsl.User@b3f21ec9`? Maybe it would be more clear what it's not happy about and what is going wrong with a proper toString or otherwise adding more logging or assertions to the test to narrow down what about the expectation is broken

So its basically comparing the data structure of https://github.com/pjfanning/incubator-pekko-connectors/blob/d52180db85f9726c0355c98eeda6639b65a660a9/slick/src/test/java/docs/javadsl/SlickTest.java#L65-L69 to the contents of what is retrieved from a Slick query, i.e. https://github.com/pjfanning/incubator-pekko-connectors/blob/d52180db85f9726c0355c98eeda6639b65a660a9/slick/src/test/java/docs/javadsl/SlickTest.java#L344-L354

In this case the contents are not equal (note that ordering does not matter since we are comparing Set's). Unfortunately the error message isn't that great because JUnit doesn't have a feature that displays the difference of the actual contents (unlike lets say ScalaTest which will show what elements are missing/extra etc etc).

**Comment by mdedetrich** [comment]
So I have just updated that branch to print out the contents of `User` (the datastructure being tested) and here are the results from [this](https://github.com/apache/incubator-pekko-connectors/actions/runs/6063736915/job/16451423699?pr=223) failing run

```
=== users is ===
id: 21 name: Name21
id: 12 name: Name12
id: 8 name: Name8
id: 30 name: Name30
id: 27 name: Name27
id: 18 name: Name18
id: 2 name: Name2
id: 36 name: Name36
id: 7 name: Name7
id: 1 name: Name1
id: 35 name: Name35
id: 20 name: Name20
id: 11 name: Name11
id: 29 name: Name29
id: 0 name: Name0
id: 38 name: Name38
id: 31 name: Name31
id: 37 name: Name37
id: 22 name: Name22
id: 13 name: Name13
id: 28 name: Name28
id: 6 name: Name6
id: 19 name: Name19
id: 33 name: Name33
id: 5 name: Name5
id: 39 name: Name39
id: 24 name: Name24
id: 15 name: Name15
id: 4 name: Name4
id: 23 name: Name23
id: 14 name: Name14
id: 32 name: Name32
id: 3 name: Name3
id: 26 name: Name26
id: 17 name: Name17
id: 10 name: Name10
id: 25 name: Name25
id: 16 name: Name16
id: 34 name: Name34
id: 9 name: Name9
=== foundUsers is ===
id: 34 name: Name34
id: 9 name: Name9
``` 

As you can see, the result of the query is just returning less results than anticipated (we should be returning the entire set)

**Comment by mdedetrich** [comment]
> Do we think this is still an issue with M5 jar? The Pekko PR for slick 3.5 passes the Slick tests after M5 uptake but this could just be luck.

Lets try running the tests a decent amount of times before we confirm this.

**Comment by mdedetrich** [comment]
So I am going to go ahead and close this. I am not sure what happened, i.e. if Slick 3.5.0-M5 fixed the issue, and/or Scala 3.3.1 and/or some changes in Pekko Connectors (highly unlikely) but this issue is no longer happening. I have updated the Slick 3.5.0 PR at https://github.com/apache/incubator-pekko-connectors/pull/223 and after many pushes + manual runs there have been over 15+ runs of the [test](https://github.com/apache/incubator-pekko-connectors/actions/runs/7771628667?pr=223) against Scala 3.3.1 and its never failed.

@nafg  Going to go ahead and close this issue
